examples/design/EU-DEMO/EUDEMO_builders/tf_coils.py

```diff
@@ -0,0 +1,828 @@
+from bluemira.geometry.tools import (
+    boolean_cut,
+    boolean_fuse,
+    extrude_shape,
+    make_polygon,
+    offset_wire,
+    slice_shape,
+    sweep_shape,
+)
+from bluemira.geometry.wire import BluemiraWire
+from bluemira.magnetostatics.circuits import (
+    ArbitraryPlanarRectangularXSCircuit,
+    HelmholtzCage,
+)
+from bluemira.utilities.optimiser import Optimiser
+from bluemira.utilities.tools import get_class_from_module
+
+
+class TFCoil(ComponentManager):
+    """
+    Wrapper around the TF Coil component tree.
+    """
+
+    def __init__(self, component, field_solver):
+        super().__init__(component)
+        self._field_solver = field_solver
+
+    def field(self, x, y, z):
+        """
+        Calculate the magnetic field due to the TF coils at a set of points.
+
+        Parameters
+        ----------
+        x: Union[float, np.array]
+            The x coordinate(s) of the points at which to calculate the field
+        y: Union[float, np.array]
+            The y coordinate(s) of the points at which to calculate the field
+        z: Union[float, np.array]
+            The z coordinate(s) of the points at which to calculate the field
+        Returns
+        -------
+        field: np.array
+            The magnetic field vector {Bx, By, Bz} in [T]
+        """
+        return self._field_solver.field(x, y, z)
+
+
+@dataclass
+class TFCoilDesignerParams(ParameterFrame):
+    """
+    TF Coil builder parameters
+    """
+
+    r_tf_current_ib: Parameter[float]
+    tk_tf_wp: Parameter[float]
+    tk_tf_wp_y: Parameter[float]
+    tf_wp_depth: Parameter[float]
+    tf_wp_width: Parameter[float]
+    tk_tf_front_ib: Parameter[float]
+    g_ts_tf: Parameter[float]
+    TF_ripple_limit: Parameter[float]
+    R_0: Parameter[float]
+    z_0: Parameter[float]
+    B_0: Parameter[float]
+    n_TF: Parameter[int]
+    tk_tf_ins: Parameter[float]
+    tk_tf_insgap: Parameter[float]
+    tk_tf_nose: Parameter[float]
+    r_tf_in: Parameter[float]
+
+
+@dataclass
+class TFCoilBuilderParams(ParameterFrame):
+    """
+    TF Coil builder parameters
+    """
+
+    R_0: Parameter[float]
+    z_0: Parameter[float]
+    B_0: Parameter[float]
+    n_TF: Parameter[int]
+    tf_wp_depth: Parameter[float]
+    tf_wp_width: Parameter[float]
+    tk_tf_front_ib: Parameter[float]
+    tk_tf_ins: Parameter[float]
+    tk_tf_insgap: Parameter[float]
+    tk_tf_nose: Parameter[float]
+    tk_tf_side: Parameter[float]
+
+
+class TFCoilDesigner(Designer[GeometryParameterisation]):
+    """
+    TF Coil Designer
+
+    Parameters
+    ----------
+    params: Union[Dict, ParameterFrame]
+        TF Coil Designer parameters
+    build_config: Dict
+        Required keys;
+
+            * param_class: str
+                A string of the import location for the parameterisation
+                class of the TF Coil
+
+        Optional keys:
+
+            * variables_map: Dict
+                variable map for the param_class
```

Comment:
Is it worth clarifying what exactly is being mapped? Is it parameter names to `param_class` constructor kwargs?